### DISCUSSION TITLE
Organization. Коды по классификаторам (ОКВЭД, ОКОПФ, ОКТМО и др.)

### DISCUSSION THREAD

**Discussion by MaximNikolaev** [fhir-ru/core#167] [element]
Добрый день!

Ресурс Organization может иметь коды по классификаторам

- ОКВЭД (должен быть список значений, с одним 'основным')
- ОКФС
- ОКОПФ
- ОКПО
- ОКТМО
- ОКАТО
- ОКОГУ

**Вопрос: Нужно ли описывать каждый из реквизитов-классификаторов (ОКВЭД, ОКФС и др.) в виде отдельных extension и codesystem?**

Описал
пример с extension-ами [https://maximnikolaev.github.io/max-fhir-ru/StructureDefinition-ru-core-organization.html](https://maximnikolaev.github.io/max-fhir-ru/StructureDefinition-ru-core-organization.html)

Представляется, что получившийся набор extension-ов можно будет использовать и в других ситуациях, а не только в профиле
организации.
В описании extension-ов можно будет добавить constraint-ы на допустимые значения и длину полей.

**Comment by Vrodvi** [comment]
Расширения необходимо определять только когда отсутствуют стандартные элементы для указания данных.
Из представленного списка:
- некоторые элементы представляют из себя уточнение типа:
ОКВЭД Общероссийский классификатор видов экономической деятельности
ОКФС общероссийский классификатор форм собственности
ОКОПФ общероссийский классификатор организационно-правовых форм
    - для указания типов имеется атрибут Organization.type
-  некоторые элементы представляют уточнение адреса:
ОКТМО Общероссийский классификатор территорий муниципальных образований
ОКАТО Общероссийский классификатор объектов административно-территориального деления
    - для указания адреса имеется атрибут Organization.address
- некоторые элементы представляют из себя идентификаторы организации: 
ОКПО Общероссийский классификатор предприятий и организаций
ОКОГУ общероссийский классификатор органов государственной власти и управления
    - для указания идентификаторов имеется атрибут Organization.identifier

Таким образом, определение указанных классификаторов в виде расширений будет противоречить идеологии стандарта.
Хотя технически такой подход возможен, но он в итоге затруднит использование стандарта, т.к. данные перестанут быть доступны в ожидаемых для них местах - идентификаторы будут не в identifier, а типизация будет осуществлена не в type.

С таким же успехом можно не использовать никакие стандарты, а просто определить собственный формат данных, в котором в плоском виде будут вперемешку перечислены все атрибуты, как это примерно и происходит в текущей реализации РФМО, где классификации организаций идут рядом с "Площадью автомобильной стоянки" (см. https://github.com/fhir-ru/core/discussions/26#discussioncomment-1758754):
![image](https://user-images.githubusercontent.com/92928137/176865507-85ee9de6-8aa4-429d-904f-f7f7be466d58.png)

Но это, конечно, ИМХО. Хотелось бы послушать и остальных участников сообщества по этой теме.


Также есть просьба по поводу "_Представляется, что получившийся набор extension-ов можно будет использовать и в других ситуациях, а не только в профиле организации._"
- можете привести примеры других ситуаций, в которых вы видите возможность переиспользования указанных extension-ов?

**Comment by MaximNikolaev** [comment]
Спасибо за ответ!

Ремарка: я рассматриваю ресурс organization как 'юридическую' сущность.
Такие данные обычно встречаются в виде карточки организации вроде https://www.list-org.com/company/1305294/.
Отсюда и попытка описать коды классификаторов как свойства самой organization через extension-ы.

По кодам в новом примере [Организация (профиль) v2](https://maximnikolaev.github.io/max-fhir-ru/StructureDefinition-ru-core-organizationv2.html):

- ОКФС</br>
  Перенесен в type.


- ОКВЭД</br>
  Есть основной и дополнительные коды ОКВЭД. В примере разделены на слайсы по значению _type.coding.system_.
  **Очевидно, что system должен быть один. В [Coding](https://build.fhir.org/datatypes.html#Coding) Не хватает свойства "основной". Есть другие варианты разделения кодов ОКВЭД на основной/дополнительные?**
_UPD 04.07.22: можно использовать расширение для обозначения 'основного' ОКВЭДа, как [https://build.fhir.org/extension-organizationaffiliation-primaryind.html](https://build.fhir.org/extension-organizationaffiliation-primaryind.html). Но тогда и другие слайсы type-а нужно будет описывать с учетом этого флага, даже те, где он не нужен. Кажется, что это громоздко и неправильно - описывать слайс например, ОКОПФа, ставя ограничение "без расширения okved-primaryInd"._

- ОКОПФ</br>
  Перенесен в type.


- ОКАТО и ОКТМО</br>
  Действительно, эти коды являются следствием из адреса организации, но
    - Я не видел систем, где такие коды были бы свойством адреса. Кажется, что в адресе такие свойства всё же держать не
      стоит. Обычно они указаны в карточке организации.
    - В элементе [Address](http://hl7.org/fhir/R4/datatypes.html#Address) нет поля для добавления этих кодов. Придётся использовать extension-ы.

  Пока перенесены в type.


- ОКОГУ</br>
  Перенесен в identifier.


- ОКПО</br>
  Слова "_...одинаковых кодов ОКПО не бывает. Случается совпадение кодов, если хозяйствующий субъект был ликвидирован, и
  тогда его код по прошествии 5 лет можно присвоить новой организации или ИП..._" привели к попытке вынести это свойство
  из состава identifier-ов.

  Перенесен в identifier.

> можете привести примеры других ситуаций, в которых вы видите возможность переиспользования указанных extension-ов?

Конкретных примеров у меня нет. Может быть в каких-то статистических report-ах?</br>
Еще была мысль, что, используя extension-ы, можно задать проверку формата в одном-единственном месте и переиспользовать
ее везде одинаково. В новом примере v2 получается, что, если в каком-то еще ресурсе сделают код по классификатору, то у него
могут задать другую проверку на формат данных.

**Reply by Vrodvi** [reply]
> Очевидно, что system должен быть один. В [Coding](https://build.fhir.org/datatypes.html#Coding) Не хватает свойства "основной". Есть другие варианты разделения кодов ОКВЭД на основной/дополнительные?

Это потому что Coding сам по себе не может быть основным/дополнительным, это не его предназначение.
"основным/дополнительным" является некоторое исходное свойство объекта, в данном случае - это вид деятельности, т.е. это тип вида деятельности "основной/дополнительный", а код вида деятельности от этого никак не меняется.
Для видов деятельности, скорее всего нужно вообще использовать отдельные ресурсы (см. https://www.hl7.org/fhir/healthcareservice.html), в которых будет типизация на основной\дополнительный и будет указан сам вид деятельности.


> В элементе [Address](http://hl7.org/fhir/R4/datatypes.html#Address) нет поля для добавления этих кодов. Придётся использовать extension-ы.

Да, это так, но если и использовать расширения, то размещать их нужно в подходящих по смыслу местах - в данном случае в адресе.
Таким образом вы достигнете и ожидаемого переиспользования - т.к. в любом месте где будет указан адрес появится возможность уточнить этот адрес по классификатору.

> Я не видел систем, где такие коды были бы свойством адреса. Кажется, что в адресе такие свойства всё же держать не стоит. Обычно они указаны в карточке организации.

Похоже, что вы путаете представление информации с её структурированием.
Каким образом выводить для представления человеку информацию, содержащуюся в ресурсах, - это отдельная задача, она не должна влиять на моделирование данных.
Даже в представленном вами примере (https://www.list-org.com/company/1305294) информация побита на блоки - не все свойства называются реквизитами организации - например, ОКВЭД вынесен отдельно в блок "Виды деятельности", что логично, и должно соответствующим образом и моделироваться. Т.е. в первую очередь, у организации есть "Виды деятельности" и уже во вторую очередь эти виды деятельность классифицированы по ОКВЭД (т.к. они могут быть классифицированы ещё и по другим справочникам).
Много чего является "свойством" организации, но это не значит что все они должны находиться непосредственно в ресурсе организация - они могут быть связаны с организацией другими способами.
Например, у организации есть руководитель - это то тоже важное свойство организации, но оно не входит в состав ресурса организация, т.к. для указания сотрудников организации есть отдельные ресурсы.

> Слова "...одинаковых кодов ОКПО не бывает. Случается совпадение кодов, если хозяйствующий субъект был ликвидирован, и тогда его код по прошествии 5 лет можно присвоить новой организации или ИП..." привели к попытке вынести это свойство из состава identifier-ов.

У идентификаторов есть период действия ( period	Σ	0..1	Period	Time period when id is/was valid for use), поэтому переиспользование в разные интервалы времени не является проблемой.

> Конкретных примеров у меня нет. Может быть в каких-то статистических report-ах?

Отчёты - это тоже вариант представления информации. Так например, имея для организации классификацию по указанным справочникам мы может собирать и представлять информацию по другим аспектам организации в разрезе данных классификаторов. Скажем собрать сводку по сотрудникам в разрезе классификации видов деятельность - для этого не нужно каждому сотруднику добавлять классификацию по ОКВЭД, эта информация наследуется из организации сотрудника.

**Comment by Vrodvi** [comment]
@eugenekogan-spb Евгений, вопрос относится к административной рабочей группе. Предлагаю её участникам подключиться к данной дискуссии.

**Reply by eugenekogan-spb** [reply]
Спасибо, @Vrodvi Виталий. Мне кажется, Вы дали исчерпывающие ответы.

В вопросе о кодировании основного/не основных ОКВЭД хотелось бы не уходить в дополнительные ресурсы. Healthcare Service, согласно описанию, - это фактический вид помощи, про который можно указать расписание, необходимость получить направление и т.п. 
Если это технически возможно, стоило бы обойтись порядком следования позиций в массиве Type. Например, решить, что первым из ОКВЭД будет идти основной ОКВЭД.
